src/dsmr/parser.h

```diff
@@ -83,7 +87,11 @@ struct ParsedData<> {
 // Do not use F() for multiply-used strings (including strings used from
 // multiple template instantiations), that would result in multiple
 // instances of the string in the binary
+#ifndef ARDUINO_ARCH_ESP8266
 static constexpr char DUPLICATE_FIELD[] PROGMEM = "Duplicate field";
+#else
+static constexpr char DUPLICATE_FIELD[] = "Duplicate field";
```

Comment:
Apparently just using F() doesn't disable string pooling for AVR and ARM
(at least for Arduino Due) as described by:

http://arduino.esp8266.com/versions/1.6.5-1106-g8253b82/doc/reference.html

And confirmed by comparing the size of the complied sketch when replacing
in parser.h:

static constexpr char INVALID_NUMBER[] PROGMEM = "Invalid number";
static constexpr char INVALID_UNIT[] PROGMEM = "Invalid unit";

with

#define INVALID_NUMBER F("Invalid number")
#define INVALID_UNIT F("Invalid unit")

Are there any reasons for not choosing this  (e.g. being a workaround
instead of a proper fix to the ESP8266 core) when there is no impact on
binary size for AVR and ARM?

2016-05-31 9:09 GMT+02:00 Ewald Wasscher <ewald@wasscher.net>:

> A slightly better workaround would be to do without string pooling and
> just use F() when building for ESP8266. Shouldn't be a problem as most
> modules have plenty of memory. I will report another issue with the ESP8266
> core though.
>
> 2016-05-31 7:45 GMT+02:00 Ewald Wasscher <ewald@wasscher.net>:
>
>> Issue #2072 was opened here:
>> https://github.com/esp8266/Arduino/issues/2072
>>
>> With regards to PROGMEM the discussion linked to below seems to make
>> sense, but I cannot say I really understand it.
>>
>>
>> http://stackoverflow.com/questions/35091862/inline-static-data-causes-a-section-type-conflict
>>
>> 2016-05-30 22:39 GMT+02:00 Matthijs Kooijman <notifications@github.com>:
>>
>>> In src/dsmr/parser.h
>>> <https://github.com/matthijskooijman/arduino-dsmr/pull/3#discussion_r65104895>
>>> :
>>>
>>> >  static constexpr char DUPLICATE_FIELD[] PROGMEM = "Duplicate field";
>>> > +#else
>>> > +static constexpr char DUPLICATE_FIELD[] = "Duplicate field";
>>>
>>> Reported a bug with the ESP core yet? Any idea what PROGMEM is defined
>>> to to cause this?
>>>
>>> —
>>> You are receiving this because you were mentioned.
>>> Reply to this email directly, view it on GitHub
>>> <https://github.com/matthijskooijman/arduino-dsmr/pull/3/files/27e2ace07f866f0285c58f57b6690c9dfc280bab#r65104895>,
>>> or mute the thread
>>> <https://github.com/notifications/unsubscribe/AB7KB-rLKWFBDDTY6BFP2q7trvl3ujbJks5qG0sRgaJpZM4Im8LH>
>>> .
>>>
>>
>>
>